assignment/final/aLife/code/master.js

```diff
@@ -0,0 +1,417 @@
+        moveset.innerHTML +=
+            `
+                <div class="row">
+                    <div class="col-sm-5 offset-1">${move.NAME}</div>
+                    <div class="col-sm-2">${move.TYPE}</div>
+                    <div class="col-sm-1" style="text-align: right;">${move.POWER != null ? move.POWER : ""}</div>
+                </div>
+            `
+        currentMoves.push({
+            NAME: move.NAME,
+            TYPE: move.TYPE,
+            POWER: move.POWER
+        });
+    }
+    return currentMoves;
+}
+
+function getCurrPkmnStats(pkmn, data) {
+    currentStats = [];
+    statSection.innerHTML =
+        `
+        <div class="col-sm-6">${data.stats[5].stat.name}: ${data.stats[5].base_stat}</div>
+        <div class="col-sm-6">${data.stats[4].stat.name}: ${data.stats[4].base_stat}</div>
+        <div class="col-sm-6">${data.stats[3].stat.name}: ${data.stats[3].base_stat}</div>
+        <div class="col-sm-6">${data.stats[2].stat.name}: ${data.stats[2].base_stat}</div>
+        <div class="col-sm-6">${data.stats[1].stat.name}: ${data.stats[1].base_stat}</div>
+        <div class="col-sm-6">${data.stats[0].stat.name}: ${data.stats[0].base_stat}</div>
+    `
+    currentStats.push({
+            STAT: data.stats[5].stat.name,
+            VALUE: data.stats[5].base_stat
+        }, {
+            STAT: data.stats[4].stat.name,
+            VALUE: data.stats[4].base_stat
+        }, {
+            STAT: data.stats[3].stat.name,
+            VALUE: data.stats[3].base_stat
+        }, {
+            STAT: data.stats[2].stat.name,
+            VALUE: data.stats[2].base_stat
+        }, {
+            STAT: data.stats[1].stat.name,
+            VALUE: data.stats[1].base_stat
+        }, {
+            STAT: data.stats[0].stat.name,
+            VALUE: data.stats[0].base_stat
+        },
+
+    );
+    return currentStats;
+}
+
+function getPkmn(x) {
+    //console.log("running in getPkmn(new pkmn)");
+    let pkmn = pkmnList.find(pkmn => pkmn.NAME === x);
+
+    fetch(baseUrl + 'pokemon/' + pkmn.ID + '/')
+        .then(res => res.json())
+        .then(data => {
+            //console.log(data);
+            pkmn.TYPE1 = data.types[0].type.name;
+            pkmn.TYPE2 = data.types.length > 1 ? data.types[1].type.name : "";
+            pkmn.SPRITES = data.sprites;
+            type1.innerHTML = pkmn.TYPE1;
+            type2.innerHTML = pkmn.TYPE2;
+            idBox.innerHTML = pkmn.ID;
+            terminalScreen.innerHTML =
+                `
+                <img src="${data.sprites.front_default}">
+                `
+            pkmn.STATS = getCurrPkmnStats(pkmn, data);
+            pkmn.MOVES = getCurrPkmnMoves(pkmn, data);
+        });
+    currentPkmn = pkmn;
+    return pkmn
+}
+
+function selectNewPkmn(x) {
+    const regex = new RegExp(".*" + pkmnName.value + ".*");
+    let count = pkmnList.filter(pkmn => pkmn.NAME.match(regex)).length;
```

Comment:
It looks like you use this filtering code a few times perhaps you should just store and reference it.  RegExs are known not to be super performant and once you have done the legwork, no need to do it again =)
```
const filteredPokemon = pkmnList.filter(pkmn => pkmn.NAME.match(regex))
````

Reply:
Also, it looks like the regex is defined a few places in the file...but you could certainly define it for your whole page.

line 267: `const regex = new RegExp(".*" + pkmnName.value + ".*");`